test/helpers/kubectl.go

```diff
@@ -1281,6 +1281,22 @@ func (kub *Kubectl) WaitforPods(namespace string, filter string, timeout time.Du
 	return err
 }
 
+// WaitForSinglePod waits up until timeout seconds have elapsed for all pods in the
```

Comment:
Just out of curiosity, what is the failure you're referring to in your commit msg?

> WaitforPods unmarshals the output of 'kubectl get pods -o json' into a
PodList object. This unmarshalling fails where there's a single pod
returned.

Reply:
I think it failed to unmarshal because the kubectl command doesn't return a JSON list, but a single item. It started working as soon as I replaced:
```diff
- podList := &v1.PodList{}
+ pod := v1.Pod{}
- err := kub.GetPods(namespace, filter).Unmarshal(podList)
+ err := kub.GetPods(namespace, filter).Unmarshal(&pod)
```